### ISSUE TITLE
Minimizer: try to remove transparent module constraints 

### ISSUE THREAD

**Issue by SkySkimmer** [JasonGross/coq-tools#148]
Typically
~~~coq
(* lots of stuff *)
Module Type T. 
Parameter x : bla.
End T.

Module M <: T.
Definition x := (something that depends on lots of stuff).
Definition y := 0.
End M.

(* error using M.y and not M.x *)
~~~
has a lot to minimize blocked by the `<:` constraint.

Real world example: https://github.com/coq/coq/pull/16967#issuecomment-1407558776

**Comment by SkySkimmer** [comment]
(meanwhile opaque constraints could be turned into Declare Module)

**Comment by JasonGross** [comment]
Duplicate of #140
